coins/bitcoin/src/bitcoin_wallet.rs

```diff
@@ -1046,30 +1046,19 @@ impl CryptoWalletBuilder<BitcoinWallet> for BitcoinWalletBuilder {
 
 impl BitcoinWalletBuilder {
     /// Allows specification of the gap limit to use for the wallet
-    pub fn with_gap_limit(&mut self, gap_limit: usize) -> &mut Self {
+    pub fn gap_limit(&mut self, gap_limit: usize) -> &mut Self {
         self.gap_limit_specified = Some(gap_limit);
         self
     }
 
     /// Allows specification of the account discovery to use for the wallet
     /// If set to false, the wallet will not search for accounts used past the first account
     /// The default is true
-    pub fn with_account_discovery(&mut self, account_discovery: bool) -> &mut Self {
+    pub fn account_discovery(&mut self, account_discovery: bool) -> &mut Self {
```

Comment:
lets have this as .account_discovery()

and .no_account_discovery()

I think boolean functions should have a function for each, it clears up the intent of the functions and makes it easier to work with.